lib/logger.js

```diff
@@ -1,4 +1,44 @@
+var lodash = require('lodash');
+var isObject = lodash.isObject;
+var has = lodash.has;
+var isString = lodash.isString;
+var isNumber = lodash.isNumber;
 var logger = require('@google/cloud-diagnostics-common').logger;
-var logConfig = process.env.GCLOUD_ERRORS_LOGLEVEL;
-var logLevel = !isNaN(logConfig) ? logConfig : logger.WARN;
-module.exports = logger.create(logLevel, '@google/cloud-errors');
+/**
+ * Creates an instance of the Google Cloud Diagnostics logger class. This
+ * instance will be configured to log at the level given by the environment or
+ * the runtime configuration property `logLevel`. If neither of these inputs are
+ * given or valid then the logger will default to logging at log level `WARN`.
+ * Order of precedence for logging level is:
+ * 1) Runtime configuration property `logLevel`
+ * 2) Environmental variable `GCLOUD_ERRORS_LOGLEVEL`
+ * 3) Default log level of `WARN`
+ * @function createLogger
+ * @param {ConfigurationOptions} initConfiguration - the desired project/error
+ *  reporting configuration. Will look for the  `logLevel` property which, if
+ *  supplied, must be a numbernumber or stringified decimal representation of a
+ *  number between and including 1 through 5
+ * @returns {Object} - returns an instance of the logger created with the given/
+ *  default options
+ */
+function createLogger (initConfiguration) {
+  // Default to log level: warn
+  var level = logger.warn;
+  if (isObject(initConfiguration) && has(initConfiguration, 'logLevel')) {
+    if (isString(initConfiguration.logLevel)) {
+      // Cast string as integer
+      level = ~~initConfiguration.logLevel;
+    } else if (isNumber(initConfiguration.logLevel)) {
+      level = initConfiguration.logLevel;
+    } else {
+      throw new Error('config.logLevel must be a number or decimal ' +
+        'representation of a number in string form');
+    }
+  } else if (has(process.env, 'GCLOUD_ERRORS_LOGLEVEL')) {
```

Comment:
We should use the environment provided log level over the config provided log level for consistency with the other agents: https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/master/index.js#L66